### PR TITLE
COMP: Prefer anonymous namespace over static template function

### PR DESCRIPTION
**Pull request by hjmjohnson**
Avoid:
warning: unused function template 'itkReadRawBytesAfterSwappingMacro' [-Wunused-template]

### CODE REVIEW

**Comment by hjmjohnson** [comment]
@dzenanz This is my attempt to fix the new compiler warning.

File: Modules/IO/RAW/include/itkRawImageIO.hxx [hunk 1]
```diff
@@ -102,7 +150,7 @@ void RawImageIO< TPixel, VImageDimension >
 }
 
 template < typename TStrongType >
-static void
+void
 itkWriteRawBytesAfterSwappingMacro( const void * buffer, std::ofstream & file, ImageIOBase::ByteOrder byteOrder,
```
Comment (dzenanz) [inline Modules/IO/RAW/include/itkRawImageIO.hxx]: The existing functions are not removed?

**Comment by hjmjohnson** [comment]
@N-Dekker Thank you. That is very helpful.  I'm working on a more complete solution.  I was trying to work around the need for a cxx as the RAW image filter previously did not have any non-templated components.  I am now making a RAWIO library.

**Comment by blowekamp** [comment]
As I recall you can make this type of template function `inline` and the warning will go away too.

**Comment by dzenanz** [comment]
If you are making non-trivial changes, you might eliminate the need for a temporary buffer when the current and target endianness are the same. Currently, a temporary buffer is created no matter whether we are writing in little endian or big endian order.

**Comment by N-Dekker** [comment]
By the way, `itkReadRawBytesAfterSwappingMacro` and `itkWriteRawBytesAfterSwappingMacro` don't look like macro's to me, but I guess you noticed that as well...  `:-)`

**Comment by hjmjohnson** [comment]
> By the way, `itkReadRawBytesAfterSwappingMacro` and `itkWriteRawBytesAfterSwappingMacro` don't look like macro's to me, but I guess you noticed that as well... `:-)`

Changed.  I was minimizing the differences from the previous code, but at this point that changes are vastly different anyway.

**Comment by hjmjohnson** [comment]
> If you are making non-trivial changes, you might eliminate the need for a temporary buffer when the current and target endianness are the same. Currently, a temporary buffer is created no matter whether we are writing in little endian or big endian order.

Done.

File: Modules/IO/RAW/include/itkRawImageIO.hxx [hunk 2]
```diff
@@ -21,8 +21,27 @@
 #include "itkRawImageIO.h"
 #include "itkIntTypes.h"
 
+
 namespace itk
 {
+
+/** Utility function for writing RAW bytes */
+extern void
+WriteRawBytesAfterSwapping( itk::ImageIOBase::IOComponentType componentType,
```
Comment (dzenanz) [inline Modules/IO/RAW/include/itkRawImageIO.hxx]: Isn't this identical re-declaration from .h file?

File: Modules/IO/RAW/include/itkRawImageIO.hxx [hunk 3]
```diff
@@ -21,8 +21,27 @@
...
+    const void * buffer,
+    std::ofstream & file,
+    itk::ImageIOBase::ByteOrder byteOrder,
+    itk::SizeValueType numberOfBytes,
+    itk::SizeValueType numberOfComponents );
+
+/** Utility function for reading RAW bytes */
+extern void
+ReadRawBytesAfterSwapping( itk::ImageIOBase::IOComponentType componentType,
```
Comment (dzenanz) [inline Modules/IO/RAW/include/itkRawImageIO.hxx]: re-declaration from .h file?

File: Modules/IO/ImageBase/src/itkRawImageIOUtilities.cxx [hunk 4]
```diff
@@ -29,15 +29,15 @@ _WriteRawBytesAfterSwappingUtility( const void * buffer, std::ofstream & file, i
 
   using InternalByteSwapperType = itk::ByteSwapper< TStrongType >;
   const itk::SizeValueType numberOfPixels = numberOfBytes / ( sizeof( TStrongType ) );
-  if ( byteOrder == itk::ImageIOBase::LittleEndian )
+  if ( byteOrder == itk::ImageIOBase::LittleEndian && InternalByteSwapperType::SystemIsBigEndian() )
   {
     TStrongType * tempBuffer = new TStrongType[numberOfPixels];
     memcpy( (char *)tempBuffer, buffer, numberOfBytes );
     InternalByteSwapperType::SwapRangeFromSystemToLittleEndian( (TStrongType *)tempBuffer, numberOfComponents );
     file.write( (char *)tempBuffer, numberOfBytes );
```
Comment (dzenanz) [inline Modules/IO/ImageBase/src/itkRawImageIOUtilities.cxx]: f

File: Modules/IO/RAW/include/itkRawImageIO.hxx (hunk 2 above)
Comment (hjmjohnson) [reply Modules/IO/RAW/include/itkRawImageIO.hxx]: Removed from .h file.  I put these extern declarations in the .hxx file as they are not intended to be used outside of this class.

**Comment by hjmjohnson** [comment]
Added windows export options for shared library builds.

**Comment by dzenanz** [comment]
The following tests FAILED:
	2257 - itkTimeVaryingBSplineVelocityFieldPointSetRegistrationTest (SEGFAULT)

**Comment by hjmjohnson** [comment]
> The following tests FAILED:
> 2257 - itkTimeVaryingBSplineVelocityFieldPointSetRegistrationTest (SEGFAULT)

I'm looking into it. That is very strange. I am very surprised as I would not think that test has anything to do with RAWIO.

**Comment by dzenanz** [comment]
The same test is segfaulting in my PR #1105, but my PR could plausibly interact with it 😄

**Comment by hjmjohnson** [comment]
> itkTimeVaryingBSplineVelocityFieldPointSetRegistrationTest

FYI the exact same hash value does not fail on my computer.:

1/1 Test #2331: itkTimeVaryingBSplineVelocityFieldPointSetRegistrationTest ...   Passed    1.55 sec

The following tests passed:
	itkTimeVaryingBSplineVelocityFieldPointSetRegistrationTest

**Pull request merged**